src/xdp/xsk.c

```diff
@@ -3371,6 +3383,82 @@ XskSockoptSetHookId(
     return Status;
 }
 
+static
+NTSTATUS
+XskSockoptSetIdealProcessor(
+    _In_ XSK *Xsk,
+    _In_ XSK_SET_SOCKOPT_IN *Sockopt,
+    _In_ KPROCESSOR_MODE RequestorMode
+    )
+{
+    NTSTATUS Status;
+    const VOID *SockoptIn;
+    UINT32 SockoptInSize;
+    UINT32 Enabled;
+    KIRQL OldIrql = {0};
+    BOOLEAN IsLockHeld = FALSE;
+
+    TraceEnter(TRACE_XSK, "Xsk=%p", Xsk);
+
+    //
+    // This is a nested buffer not copied by IO manager, so it needs special care.
+    //
+    SockoptIn = Sockopt->InputBuffer;
+    SockoptInSize = Sockopt->InputBufferLength;
+
+    if (SockoptInSize < sizeof(Enabled)) {
+        Status = STATUS_BUFFER_TOO_SMALL;
+        goto Exit;
+    }
+
+    __try {
+        if (RequestorMode != KernelMode) {
+            ProbeForRead((VOID*)SockoptIn, SockoptInSize, PROBE_ALIGNMENT(UINT32));
+        }
+        RtlCopyVolatileMemory(&Enabled, SockoptIn, sizeof(UINT32));
```

Comment:
You used `sizeof(Enabled)` above instead of `sizeof(UINT32)` here. I'd recommend being consistent.

Reply:
Yeah, I tried being consistent with the line two lines up, which can't be Enabled because the compiler doesn't support it. I think your suggestion is more consistent and less dangerous.